src/Microsoft.Diagnostics.EventFlow.Inputs.ActivitySource/ActivitySourceInput.cs

```diff
@@ -0,0 +1,260 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Validation;
+
+using Microsoft.Diagnostics.EventFlow.Configuration;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs.ActivitySource
+{
+    public class ActivitySourceInput : IObservable<EventData>, IDisposable
+    {
+        private const int SamplingDecisionCacheFlushThreshold = 1024;
```

Comment:
Now I see one reason you implemented in this way, i.e., to support unrestricted ActivitySource while keep the matching in the order they are configured. You even documented the case matching multiple configurations.
I wonder if it makes sense to check such case during initialization and fail if user put a more general condition before others.

Reply:
The main reason for implementing the sampling decision cache the way it is was performance. I want the `OnActivityStarted` and `OnActivityStopped` to be really fast in the average case. 

A fixed-size, concurrent dictionary that would purge the least recently used entries might be ideal, and I even did some investigation related to it, but it involved complexity/3rd party code that I was not convinced was worth the extra trouble. So I ended up with a simple cache reset logic, with the assumption that the number of often-occurring activities will be limited and the decision for them will be cached. Even if occasional flush occurs, the cache will be quickly re-populated for often-occurring activities.

Reply:
The suggestion to warn about the case where `(Source, *)` configuration precedes `(Source, ActivityName)` is a good one, because the latter will never be used. I will add that warning and a test for it.

Reply:
Oh, I didn't mean LRU cache with fixed size number of entries. By fixed size I mean the size of the dictionary will be fixed if we construct the dictionary directly from the configuration.
Imagine the case where there is no "unrestricted activity source name", if user specified 5 configurations with explicit ActivitySourceName and ActivityName, then the size of dictionary will be 5.
But then I realized in the case to support "unrestricted configuration" it won't work because the order of unrestricted setting vs explicit setting matters. And thus said if we don't allow ```(Source, *)``` to precede ```(Source, ActivityName)```, i.e., throw exception instead of warning upfront, then we never need to flush the dictionary.